File: relationship-manager/auth-service/src/main/java/com/bank/rm/auth/dto/AuthDtos.java
package com.bank.rm.auth.dto;

import java.time.Instant;
import java.util.UUID;

public class AuthDtos {

    public record RegisterRequest(String email, String password, String name, String phone) {}

    public record LoginRequest(String email, String password) {}

    public record ConvertSessionRequest(UUID anonymousSessionId, String email, String password, String name) {}

    public record AuthResponse(UUID userId, String token, String email, String name, Instant expiresAt) {}

    public record AnonymousSessionResponse(UUID sessionId, String token) {}

    private AuthDtos() {}
}

File: relationship-manager/auth-service/src/main/kotlin/com/bank/rm/auth/dto/AuthDtos.kt
package com.bank.rm.auth.dto

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size
import java.util.UUID

data class RegisterRequest(
    @field:Email @field:NotBlank
    val email: String,
    @field:NotBlank @field:Size(min = 8, max = 128)
    val password: String,
    val phone: String? = null,
    val anonymousSessionId: UUID? = null
)

data class LoginRequest(
    @field:Email @field:NotBlank
    val email: String,
    @field:NotBlank
    val password: String
)

data class AuthResponse(
    val userId: UUID,
    val token: String,
    val refreshToken: String,
    val profileMigrated: Boolean = false,
    val conversationsMigrated: Int = 0
)

data class RefreshRequest(
    @field:NotBlank
    val refreshToken: String
)

data class TokenResponse(
    val token: String,
    val refreshToken: String
)
